File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.java
class Solution {
    public int countMajoritySubarrays(int[] nums, int target) {
        int count = 0;

        for (int i = 0; i < nums.length; i++){
            int curr = 0;

            for (int j = i; j < nums.length; j++){
                if (nums[j] == target) curr += 1;
                else curr -= 1;

                if (curr > 0) count += 1;
            }
        }

        return count;
    }
}

File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.py
class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        count = 0

        for i in range(len(nums)):
            curr = 0

            for j in range(i,len(nums)):
                curr += 1 if nums[j] == target else -1
                count += 1 if curr > 0 else 0
                
        return count